multicluster/cmd/multicluster-controller/gateway_webhook.go

```diff
@@ -59,6 +59,14 @@ func (v *gatewayValidator) Handle(ctx context.Context, req admission.Request) ad
 		klog.ErrorS(err, "failed to create Gateway", "Gateway", klog.KObj(gateway), "Namespace", v.namespace)
 		return admission.Errored(http.StatusPreconditionFailed, err)
 	}
+	if req.Operation == admissionv1.Update && len(gatewayList.Items) > 0 {
+		oldGateway := gatewayList.Items[0]
+		if oldGateway.ServiceCIDR != gateway.ServiceCIDR {
```

Comment:
This is not correct, you should compare the req.OldObject and req.Object's ServiceCIDR instead of comparing gatewayList.Items[0]'s CIDR with current Gateway's.

Reply:
@luolanzone : I do not understand why we need these checks if Gateway is created by MC Controller. The webhook is generally for validating user intents?

If the goal is to prevent users from modifying the object, we should just check that, and ideally override user changes at controller restart.

Reply:
Yeah, I think the purpose is to prevent users to modify the field. @jianjuns do you mean not to add validation on webhook but check the field and override the change in node_controller?

Reply:
Ideally MC Controller should check and override changes. But first it makes no sense to validate particular fields here. If we want to prevent modification from users, can we just check the identity?

Reply:
Good point, @hjiajing  could you refine this part? Thanks.

Reply:
Sure, I will refine this part.

Reply:
Now the controller will check the SA if the verb is "update"